Doc/library/signal.rst

```diff
@@ -312,16 +312,36 @@ The :mod:`signal` module defines the following functions:
    If not -1, *fd* must be non-blocking.  It is up to the library to remove
    any bytes from *fd* before calling poll or select again.
 
-   Use for example ``struct.unpack('%uB' % len(data), data)`` to decode the
-   signal numbers list.
-
    When threads are enabled, this function can only be called from the main thread;
    attempting to call it from other threads will cause a :exc:`ValueError`
    exception to be raised.
 
+   There are two common ways to use this function. In both approaches,
+   you use the fd to wake up when a signal arrives, but then they
+   differ in how they determine *which* signal or signals have
+   arrived.
+
+   In the first approach, we read the data out of the fd's buffer, and
+   the byte values give you the signal numbers. This is simple, but it
+   can run into a problem: generally the fd will have a limited amount
+   of buffer space, and if too many signals arrive too quickly, the
+   buffer may become full, and some signals may be lost. If you use
```

Comment:
I propose to add something like "if many signals are received quickly, faster that what the application can read".

I never saw this warning, there is need to scare users ;-)

Reply:
Tweaked the text to emphasize that this is a rare condition.